config/v1/types_network.go

```diff
@@ -13,20 +13,56 @@ type Network struct {
 	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// spec holds user settable values for configuration
+	// spec holds user settable values for configuration.
 	Spec NetworkSpec `json:"spec"`
 	// status holds observed values from the cluster. They may not be overridden.
 	Status NetworkStatus `json:"status"`
 }
 
+// NetworkSpec is the desired network configuration.
+// As a general rule, this SHOULD NOT be read directly. Instead, you should
+// consume the NetworkStatus, as it indicates the currently deployed configuration.
+// Currently, none of these fields may be changed after installation.
 type NetworkSpec struct {
-	// serviceCIDR
-	// servicePortRange
-	// vxlanPort
-	// 	ClusterNetworks    []ClusterNetworkEntry `json:"clusterNetworks"`
+	// IP address pool to use for pod IPs.
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	ServiceNetwork []string `json:"serviceNetwork"`
+
+	// NetworkType is the plugin that is to be deployed (e.g. OpenShiftSDN).
+	// This should match a value that the cluster-network-operator understands,
+	// or else no networking will be installed.
+	// Currently supported values are:
+	// - OpenShiftSDN
+	NetworkType string `json:"networkType"`
 }
 
+// NetworkStatus is the current network configuration.
 type NetworkStatus struct {
+	// IP address pool to use for pod IPs.
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
+
+	// IP address pool for services.
+	// Currently, we only support a single entry here.
+	ServiceNetwork []string `json:"serviceNetwork"`
+
+	// NetworkType is the plugin that is deployed (e.g. OpenShiftSDN).
+	NetworkType string `json:"networkType"`
+
+	// PodNetworkMTU is the discovered MTU for inter-pod networking.
+	PodNetworkMTU int `json:"podNetworkMTU"`
```

Comment:
I'd call it ClusterNetworkMTU for consistency. Also, presumably if the admin overrides it, we're going to show the actual, overridden MTU here, not the incorrect "discovered" MTU, so the doc is wrong.

Reply:
good point, updated.

Reply:
> good point, updated.

you have to actually type "git push" after you say that ;-)

Reply:
works on my machine ;-p